### PR TITLE
improve print_model_weights output by adding a new argument to sort feature weights by sort_by_labels #442

### PR DESCRIPTION
**Pull request by bndgyawali**
This PR improves `print_model_weights` by adding a new argument `--sort_by_labels` which when true sorts the features by classes. By default, it is false. 

This new argument is mutually exclusive to `--k` which mean it only works when `--k` is not specified.

### CODE REVIEW

**Comment by bndgyawali** [comment]
I am going to add a new test case. So, please hold a while to review it.

File: skll/utilities/print_model_weights.py [hunk 1]
```diff
@@ -58,6 +64,13 @@ def main(argv=None):
     learner = Learner.from_file(args.model_file)
     (weights, intercept) = learner.model_params
 
+    multiclass = False
+    model = learner._model
+    if (isinstance(model, LinearSVC) or
+        isinstance(model, LogisticRegression) or
+        (isinstance(model, SVC) and
+            model.kernel == 'linear')):
+        multiclass = True
```
Comment (desilinguist) [inline skll/utilities/print_model_weights.py]: You could be using `LogisticRegression` for bianary classification in which case you will only have weights for one class and then you don't need to do this. May be you need an condition that the number of classes (via `self.label_list`) is  > 2?

File: skll/utilities/print_model_weights.py [hunk 2]
```diff
@@ -69,19 +82,28 @@ def main(argv=None):
...
     print("Number of nonzero features:", len(weights), file=sys.stderr)
-    for feat, val in sorted(weight_items, key=lambda x: -abs(x[1]))[:k]:
-        print("{:.12f}\t{}".format(val, feat))
+    weight_by_class = defaultdict(dict)
+    if multiclass and args.sort_by_labels:
+        for label_feature, weight in weight_items:
+            label, feature = label_feature.split()
+            weight_by_class[label][feature] = weight
+        for label in sorted(weight_by_class):
+            for feat, val in sorted(weight_by_class[label].items(), key=lambda x: -abs(x[1])):
```
Comment (desilinguist) [inline skll/utilities/print_model_weights.py]: You need to show a warning to the suer that the `k` part is ignored when you are sorting.

**Comment by bndgyawali** [comment]
I have addressed the comments and added extra tests to test the output of `print_model_weights` when `sort_by_labels` is passed as an argument. 
And, it's ready for review.

**Comment by desilinguist** [comment]
@jbiggsets @mulhod if you get a chance, please review this PR.

**Pull request merged**